cranelift-codegen/meta/src/shared/immediates.rs

```diff
@@ -76,45 +76,40 @@ pub(crate) struct Immediates {
 impl Immediates {
     pub fn new() -> Self {
         Self {
-            imm64: Builder::new_imm("imm64")
+            imm64: Builder::new_imm("imm", "ir::immediates::Imm64")
                 .doc("A 64-bit immediate integer.")
                 .build(),
 
-            uimm8: Builder::new_imm("uimm8")
+            uimm8: Builder::new_imm("imm", "ir::immediates::Uimm8")
                 .doc("An 8-bit immediate unsigned integer.")
                 .build(),
 
-            uimm32: Builder::new_imm("uimm32")
+            uimm32: Builder::new_imm("imm", "ir::immediates::Uimm32")
                 .doc("A 32-bit immediate unsigned integer.")
                 .build(),
 
-            uimm128: Builder::new_imm("uimm128")
+            uimm128: Builder::new_imm("imm", "ir::Immediate")
                 .doc("A 128-bit immediate unsigned integer.")
-                .rust_type("ir::Immediate")
                 .build(),
 
-            pool_constant: Builder::new_imm("poolConstant")
+            pool_constant: Builder::new_imm("constant_handle", "ir::Constant")
                 .doc("A constant stored in the constant pool.")
-                .format_field_name("constant_handle")
-                .rust_type("ir::Constant")
                 .build(),
 
-            offset32: Builder::new_imm("offset32")
+            offset32: Builder::new_imm("offset", "ir::immediates::Offset32")
```

Comment:
I see what you mean by this being very explicit but this level of explicitness would have really helped me when I was getting started.